sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/DiagnosticDescriptorBuilder.cs

```diff
@@ -62,6 +65,25 @@ private static IEnumerable<string> BuildCustomTags(string diagnosticId, Resource
                 yield return SonarWayTag;
             }
 
+            var scope = resourceManager.GetString($"{diagnosticId}_Scope");
+            if (scope == "Main")
+            {
+                yield return MainSourceScopeTag;

```

Comment:
Using _yield_ doesn't really add anything here; it doesn't make the code more readable, but it does add to the execution time.